index.js

```diff
@@ -0,0 +1,126 @@
+var PARAMS = /\{([^\}]*)\}/g;
+var PROPS = /\{\{([^\}]*)\}\}/g;
+
+var crossroads = require('crossroads');
+var objectpath = require('object-path');
+var humanize = require('string-humanize');
+var path = require('path');
+var strip = require('strip');
+var urlRegex = require('url-regex');
+
+function parse( content, context ){
+  if( !content ) return false;
+
+  // Files
+  if( content.split('.').pop().match(/gif|jpg|jpeg|png/) ){
+    return content
+      ? path.join(context.url.origin, content)
+      : false;
+  }
+
+  // URLs
+  if( content && content.slice(0,4) === 'http' ){
+    return urlRegex({ exact: true }).test(content)
+      ? content
+      : false;
+  }
+
+  ///////////////////
+  // context data //
+  /////////////////
+
+  if( content.match(/\./) && (!content.match(/\s/) || content.match(/\{{/)) ){
+    // Normalize array selection syntax
+    content = content.replace(/\[|\]/g,'.').replace(/\.\./g,'.');
+
+    var props = content.match(PROPS);
+
+    if( props ){
+      props.forEach(function( prop ){
+        var key = (prop.substr(2, prop.length - 4)).trim();
+        var data = objectpath.get(context, key);
+        if( data ) content = content.replace(prop, data);
+      });
+    } else {
+      var content = objectpath.get(context, content);
+    }
+
+    return content
+      ? content
+      : false;
+  }
+
+  return content;
+}
+
+function setCanonical( originPath, context ){
+  var params = originPath.match(PARAMS) || [];
+
+  params.forEach(function( parameter ){
+    var value = context.parameters[parameter.slice(1,-1)];
+    originPath = originPath.replace(parameter, value);
+  });
+
+  return path.join(context.url.origin, originPath);
+}
+
+module.exports = function( context, metadata ){
+
+  var metadataRouter = crossroads.create();
+
+  // Sort routes to prioritize deeper routes
+  var routes = Object.keys(metadata).sort(function( a, b ){
+    return b.match(/\//g).length > a.match(/\//g).length
```

Comment:
You could also match static routes first, else they get overridden by dynamic routes. Or we let the user deal with routing priority completely?